src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDirective.java

```diff
@@ -85,16 +86,25 @@ void putMetric(String key, double value, Unit unit, StorageResolution storageRes
         metrics.compute(
                 key,
                 (k, v) -> {
-                    if (v == null) return new MetricDefinition(key, unit, storageResolution, value);
-                    else {
-                        v.addValue(value);
+                    if (v == null) {
+                        MetricDefinition.MetricDefinitionBuilder builder =
+                                MetricDefinition.builder()
+                                        .name(k)
+                                        .unit(unit)
+                                        .storageResolution(storageResolution)
+                                        .addValue(value);
+                        return builder;
+                    } else if (v instanceof Metric.MetricBuilder) {
+                        ((Metric.MetricBuilder) v).addValue(value);
                         return v;
+                    } else {
+                        throw new InvalidMetricException("Metric already exists and is Immutable");
```

Comment:
Can we adjust the exception message to include information on how to recover. Assume you are getting this exception when building your project, what is your next step?